### PR TITLE
Feature/added missing classes

### PR DESCRIPTION
**Pull request by agabylicka**
Added missing classes and tests.

### CODE REVIEW

**Comment by blazejprzyluski** [comment]
In the future please make the commit messages more descriptive

File: src/main/java/org/example/ClassSpells.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+
+public class ClassSpells {
+    public final String userURL = "https://www.dnd5eapi.co/api/classes/:index/levels/1/spells";
```
Comment (blazejprzyluski) [inline src/main/java/org/example/ClassSpells.java]: why is it public? why the userURL? Is it related to users?

File: src/main/java/org/example/ClassSpells.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    private final HttpClient client = HttpClient.newHttpClient();
+
+    public HttpResponse<String> getClassSpells(String index) {
+        HttpRequest request;
+        try {
+            request = HttpRequest.newBuilder(new URI(userURL.replace(":index", index))).GET().build();
+            HttpResponse<String>response = client.send(request,HttpResponse.BodyHandlers.ofString());
+            return response;
+        } catch (URISyntaxException e) {
```
Comment (blazejprzyluski) [inline src/main/java/org/example/ClassSpells.java]: all those catch blocks have the same response. Please create separate responses or just a single catch block

File: src/main/java/org/example/Proficiencies.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+
+public class Proficiencies {
+    public final String userURL = "https://www.dnd5eapi.co/api/races/:index/proficiencies";
```
Comment (blazejprzyluski) [inline src/main/java/org/example/Proficiencies.java]: why public? also same question about naming, how is it related to users?

File: src/main/java/org/example/Proficiencies.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    public final HttpClient client = HttpClient.newHttpClient();
```
Comment (blazejprzyluski) [inline src/main/java/org/example/Proficiencies.java]: why public?

File: src/main/java/org/example/Proficiencies.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    public HttpResponse<String>getProficiencies(String index) {
```
Comment (blazejprzyluski) [inline src/main/java/org/example/Proficiencies.java]: nitpick: please introduce spaces in method declaration (ctyrl + alt + l)

File: src/main/java/org/example/Proficiencies.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+        HttpRequest request;
+        try {
+            request = HttpRequest.newBuilder(new URI(userURL.replace(":index", index))).GET().build();
+            HttpResponse<String>response = client.send(request,HttpResponse.BodyHandlers.ofString());
```
Comment (blazejprzyluski) [inline src/main/java/org/example/Proficiencies.java]: same here with spaces

File: src/main/java/org/example/Proficiencies.java [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+            return response;
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
```
Comment (blazejprzyluski) [inline src/main/java/org/example/Proficiencies.java]: same issue with catch blocks, please collapse them or handle those errors separately

File: src/main/java/org/example/RaceInformation.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+            return response;
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        } catch (URISyntaxException e) {
+            throw new RuntimeException(e);
+        }
+    }
+}
```
Comment (blazejprzyluski) [inline src/main/java/org/example/RaceInformation.java]: Same comments as to other classes handling api

File: src/test/java/TestClassFeatures.java [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+
+public class TestClassFeatures {
+    @Test
+    public void testClassFeatures() {
+        String index = "druid";
+        ClassFeatures classFeatures = new ClassFeatures();
+        String expected = "druid";
+
+        HttpResponse<String> actual = classFeatures.getClassFeatures(index);
+        Assertions.assertTrue(actual.body().contains(expected));
```
Comment (blazejprzyluski) [inline src/test/java/TestClassFeatures.java]: very weak test

File: src/test/java/TestClassFeatures.java [hunk 10]
```diff
@@ -0,0 +1,18 @@
+import org.example.ClassFeatures;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.net.http.HttpResponse;
+
+public class TestClassFeatures {
```
Comment (blazejprzyluski) [inline src/test/java/TestClassFeatures.java:7]: please introduce more cases

**Review by blazejprzyluski — COMMENTED** [review]
Please add more test cases, negative paths etc. No need to do mockito yet, we can discuss it later on

File: src/main/java/org/example/ClassSpells.java (hunk 2 above)
Comment (blazejprzyluski) [reply src/main/java/org/example/ClassSpells.java]: IOException | InterruptedException

File: src/main/java/org/example/RaceInformation.java [hunk 11]
```diff
@@ -0,0 +1,24 @@
+package org.example;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+
+//"https://www.dnd5eapi.co/api/races/:index/proficiencies"; GET /api/races/:index
```
Comment (blazejprzyluski) [inline src/main/java/org/example/RaceInformation.java]: unnecessary comment

File: src/main/java/org/example/RaceInformation.java [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+public class RaceInformation {
+    private final String userURL = "https://www.dnd5eapi.co/api/races/:index";
```
Comment (blazejprzyluski) [inline src/main/java/org/example/RaceInformation.java]: ctrl + alt + l

File: src/test/java/TestClassSpells.java [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+import org.junit.jupiter.api.Test;
+
+import java.net.http.HttpResponse;
+
+public class TestClassSpells {
+        @Test
+        public void testClassSpells() {
+            String index = "bard";
+            ClassSpells classSpells = new ClassSpells();
+            //String expected = "Bane";
```
Comment (blazejprzyluski) [inline src/test/java/TestClassSpells.java]: to be cleaned up

File: src/main/java/org/example/ClassSpells.java (hunk 1 above)
Comment (agabylicka) [reply src/main/java/org/example/ClassSpells.java]: Fixed

File: src/main/java/org/example/ClassSpells.java (hunk 2 above)
Comment (agabylicka) [reply src/main/java/org/example/ClassSpells.java]: Done

File: src/main/java/org/example/Proficiencies.java (hunk 3 above)
Comment (agabylicka) [reply src/main/java/org/example/Proficiencies.java]: Fixed

File: src/main/java/org/example/Proficiencies.java (hunk 4 above)
Comment (agabylicka) [reply src/main/java/org/example/Proficiencies.java]: Fixed

File: src/main/java/org/example/Proficiencies.java (hunk 5 above)
Comment (agabylicka) [reply src/main/java/org/example/Proficiencies.java]: Done

File: src/main/java/org/example/Proficiencies.java (hunk 6 above)
Comment (agabylicka) [reply src/main/java/org/example/Proficiencies.java]: Done

File: src/main/java/org/example/Proficiencies.java (hunk 7 above)
Comment (agabylicka) [reply src/main/java/org/example/Proficiencies.java]: Done

File: src/main/java/org/example/RaceInformation.java (hunk 8 above)
Comment (agabylicka) [reply src/main/java/org/example/RaceInformation.java]: Done

File: src/main/java/org/example/RaceInformation.java (hunk 11 above)
Comment (agabylicka) [reply src/main/java/org/example/RaceInformation.java]: Done

File: src/main/java/org/example/RaceInformation.java (hunk 12 above)
Comment (agabylicka) [reply src/main/java/org/example/RaceInformation.java]: Done

File: src/test/java/TestClassSpells.java (hunk 13 above)
Comment (agabylicka) [reply src/test/java/TestClassSpells.java]: Done

File: src/test/java/TestClassFeatures.java (hunk 9 above)
Comment (agabylicka) [reply src/test/java/TestClassFeatures.java]: Fixed

File: src/test/java/TestClassFeatures.java (hunk 10 above)
Comment (agabylicka) [reply src/test/java/TestClassFeatures.java:7]: Fixed
